### PR TITLE
Sentieon update

### LINKED ISSUE #128: Update Sentieon to 2023 version (or check that the results are identical) - Latest February

### PR DESCRIPTION
**Pull request by Jakob37**
# Description

* Update of Sentieon container to 202308 version (build based on https://github.com/Sentieon/sentieon-docker)
* Put the container as a params
* Split the Sentieon QC process and update the used Perl script (such that Sentieon can be run in a separate container)

### Reviewer considerations

The most sensitive parts are:

* The moving of a Sentieon driver command from the config into the main.nf script
* The splitting up of the QC process into two processes, and the communication between these

It would be great if these parts were carefully eyed through by another pair of eyeballs.

FIXME:

- [x] Document in the docs how the container was generated
- [x] Sentieon_qc: Contains two commands - one for Sentieon, and  `qc_sentieon.pl`. The latter crashed due to missing `Data/Dumper.pm` dependency, one solution would be to split that process into two
- [x] Strange stub errors from AnnotSV claiming: " Missing output file(s) `group/group.merged.bndless.annotated.tsv` expected by process `annotsv (group)`"
- [x] Update software versions in the docs

Close #128 

## Type of change

- [x] Minor change

# Checklist:

- [x] My code follows the [style guidelines for NF pipelines at CMD](http://mtlucmds1.lund.skane.se/wiki/doku.php?id=nextflow&s[]=nextflow#code_style_at_cmd)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings (Keep an eye on `.nextflow.log` !)
- [x] I have updated the CHANGELOG
- [x] The latest commit in the master branch is tagged
      with the correct version number

## Minor change
- [x] Stub run completes without errors or new warnings
- [/] `onco` run finishes without any new warnings/errors and the results can 
       be loaded into scout (ran but not loaded into Scout yet)
- [x] `wgs` single run finishes without any new warnings/errors and the results 
      can be loaded into scout
- [x] `wgs` trio run finishes without any new warnings/errors and the results 
       can be loaded into scout
- [x] At least one other person has reviewed and approved my code
- [x] I have made corresponding changes to the documentation
 
# Post-merge

- [x] The `master` branch has been tagged with the new version number.

### CODE REVIEW

**Comment by alkc** [comment]
+1 for chucking `qc_sentieon.pl` into its own process. That script needs to be gutted and replaced anyway (#80), having it banished/exiled to its own process will make it easier to switch it out.

**Comment by Jakob37** [comment]
Onco run QC output looks identical in format (and differing in a handful of mapped reads in the values).

I'll testrun and check the wgs and wgs trio as well

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,28 +24,22 @@
 * changed path to normal-pool-refs for gens. Uses masked hg38 references
 
 ### 3.5.9
-
```
Comment (Jakob37) [inline CHANGELOG.md:21]: The inconsistency was irritating my eyes

**Comment by Jakob37** [comment]
Now it is testrun with wgs single/trio and one onco sample. It runs without any errors and they could be loaded into Scout (into @alkc test instance).

I think this is ready for further investigations now!

File: main.nf [hunk 2]
```diff
@@ -533,6 +541,43 @@ def sentieon_qc_version(task) {
...
+			--dedup_metrics_file ${dedup} \\
+			--metrics_file ${assay_metrics} \\
+			--gcsummary_file ${gc_summary} \\
+			--coverage_file ${cov_metrics} \\
+			--coverage_file_summary ${cov_metrics_sample_summary} \\
+			> ${id}_qc.json
+		
+		"""
+}
+
```
Comment (alkc) [inline main.nf:580]: Heart on fire emoji

**Review by alkc — APPROVED** [review]
Nothing to add. The changes look good to me!

File: main.nf (hunk 2 above)
Comment (alkc) [reply main.nf:580]: Regarding the rewrite of the process + qc script.

**Pull request merged**